0.110.0/rockcraft.yaml

```diff
@@ -0,0 +1,54 @@
+# Based on: https://github.com/open-telemetry/opentelemetry-collector-releases/blob/main/distributions/otelcol/Dockerfile
+name: opentelemetry-collector
+summary: Crucial component in the OpenTelemetry ecosystem.
+description: |
+  Designed to collect, process, and export telemetry data such as metrics, logs,
+  and traces from various sources to multiple destinations.
+version: "0.110.0"
+base: ubuntu@24.04
+license: Apache-2.0
+platforms:
+  amd64:
```

Comment:
Upstream [cross-compiles](https://github.com/open-telemetry/opentelemetry-collector-releases/blob/f58f0d245a7164cbcaa74448f773a8e3c958a150/.github/workflows/base-release.yaml#L25) to arm64 on amd64. Not sure where rockcraft, oci-factory etc. currently stand with multiarch images, but at some point we'd need an arm64 image.
We can tackle this in a separate issue.

Reply:
See https://github.com/canonical/opentelemetry-collector-rock/issues/2